Source/utils/file_util.cpp

```diff
@@ -3,6 +3,7 @@
 
 #include <algorithm>
 #include <string>
+#include <sys/stat.h>

```

Comment:
I'm surprised this works on all platforms that we support!

Reply:
Positively surprised I assume :D

Reply:
Please move the `<sys/stat.h>` include back into the `#if _POSIX_C_SOURCE >= 200112L` block